lxd/images.go

```diff
@@ -254,14 +253,14 @@ func getImgPostInfo(d *Daemon, r *http.Request, builddir string) (info shared.Im
 	var size int64
 
 	// Create a temporary file for the image tarball
-	imageTarf, err := ioutil.TempFile(builddir, "lxd_tar_")
+	imageTarf, err := ioutil.TempFile("", "lxd_tar_")
 	if err != nil {
 		return info, err
 	}
 
 	if ctype == "multipart/form-data" {
 		// Create a temporary file for the rootfs tarball
-		rootfsTarf, err := ioutil.TempFile(builddir, "lxd_tar_")
+		rootfsTarf, err := ioutil.TempFile("", "lxd_tar_")
```

Comment:
So I'm not sure I agree with all those changes from using a subdir of builddir to using /tmp as the temporary area. Sure the new FileMove function will avoid complete failure, but it's still going to be quite a lot more overhead due to copy+delete, so reading and writing every single block when not on the same filesystem.

Maybe we should have a tmpdir which is a sub-dir of /var/lib/lxd? that would make it more likely to be on the same filesystem and so make move operations instantaneous again.